flink-clients/src/main/java/org/apache/flink/client/deployment/executors/RemoteExecutor.java

```diff
@@ -1,38 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.flink.client.deployment.executors;
-
-import org.apache.flink.annotation.Internal;
-import org.apache.flink.client.deployment.AbstractSessionClusterExecutor;
-import org.apache.flink.client.deployment.StandaloneClientFactory;
-import org.apache.flink.client.deployment.StandaloneClusterId;
-import org.apache.flink.core.execution.Executor;
-
-/**
- * The {@link Executor} to be used when executing a job on an already running cluster.
- */
-@Internal
-public class RemoteExecutor extends AbstractSessionClusterExecutor<StandaloneClusterId, StandaloneClientFactory> {
```

Comment:
Actually I think `RemoteExecutor` is more general than `StandaloneSessionExecutor`. Because in Scala shell, we actually use the executor for connecting Flink Cluster on YARN. The `StandaloneClientFactory` doesn't care about which deployment mode the cluster is, but if only there is a Flink cluster.

Reply:
↑ name issue & document issue.

Also StandaloneSessionExecutor says it is "A {@link ClusterClientFactory} for a standalone cluster, i.e. Flink on bare-metal.", but still, we use it in Scala Shell for connecting Flink Cluster on YARN.

Reply:
Currently the `ScalaShell` for `Yarn` uses the `YarnSessionClusterExecutor` but I get the point that `RemoteExecutor` seems a more intuitive name.

So you suggest to remove the `StandaloneSessionClusterExecutor` in favour of the `RemoteExecutor` instead of the other way round. Right?

I will do that.